### PR TITLE
feat!: one time event listeners

### PR DESCRIPTION
**Pull request by Om1609**
## Summary

Reverts #1940 and reimplements one time event listeners using `@client.listen(once=True)`.

Breaking change for master branch. The code this pr breaks isn't released as of making this pr, but merged with master branch.

## Information

- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [ ] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [ ] I have updated the changelog to include these changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,8 +15,8 @@ These changes are available on the `master` branch, but have not yet been releas
 - Added new events `on_bridge_command`, `on_bridge_command_completion`, and
   `on_bridge_command_error`.
   ([#1916](https://github.com/Pycord-Development/pycord/pull/1916))
-- Added the `@client.once()` decorator, which serves as a one-time event listener.
-  ([#1940](https://github.com/Pycord-Development/pycord/pull/1940))
+- Added support for one-time event listeners in `@client.listen()`.
```
Comment (Lulalaby) [inline CHANGELOG.md:18]: Please split to removed and added

**Review by Lulalaby — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Om1609) [reply CHANGELOG.md:18]: For unreleased (stable) changes too? That would be misleading

Comment (Lulalaby) [reply CHANGELOG.md:18]: You can't just override a while change like that
Even if it wasn't released yet, users were able to install it via git@master

Comment (Om1609) [reply CHANGELOG.md:18]: Alright cool. On it

**Review by Lulalaby — APPROVED** [review]

**Review by VincentRPS — DISMISSED** [review]
lgtm

**Review by Lulalaby — APPROVED** [review]

**Review by ChickenDevs — APPROVED** [review]

**Pull request merged**